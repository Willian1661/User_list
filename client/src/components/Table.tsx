import React from "react";
import User from "../interfaces/user";
interface props {
    data: Promise<User[]>
    title: string
}
const Table: React.FC<props> = ({ data, title }) => {
    const [tableData, setTableData] = React.useState<User[]>([]);

    React.useEffect(() => {
        const fetchData = async () => {

            const resolvedData = await data;
            setTableData(resolvedData);

        };
        fetchData();
    }, [data]);

    return (
        <div className="flex items-center h-screen">
            <div className=" px-1 min-w-max">
                <table id="table" className=" border-collapse border-2 border-solid">
                    <caption className=" text-2xl">
                        {title}
                    </caption>
                    <thead>
                        <tr>
                            <th>ID</th>
                            <th>Name</th>
                            <th>UserName</th>
                            <th>Email</th>
                            <th>phone</th>
                            <th>website</th>
                            <th>address(city)</th>
                            <th>address(zipcode)</th>
                            <th>address(street)</th>
                            <th>address(suite)</th>
                            <th>address geo(lat)</th>
                            <th>address geo(lng)</th>
                            <th>company(name)</th>
                            <th>company(catchPhrase)</th>
                            <th>company(bs)</th>
                        </tr>
                    </thead>
                    <tbody>
                        {tableData.map(user => (
                            <tr key={user.id}>
                                <td>{user.id}</td>
                                <td>{user.name}</td>
                                <td>{user.username}</td>
                                <td>{user.email}</td>
                                <td>{user.phone}</td>
                                <td>{user.website}</td>
                                <td>{user.address.city}</td>
                                <td>{user.address.zipcode}</td>
                                <td>{user.address.street}</td>
                                <td>{user.address.suite}</td>
                                <td>{user.address.geo.lat}</td>
                                <td>{user.address.geo.lng}</td>
                                <td>{user.company.name}</td>
                                <td>{user.company.catchPhrase}</td>
                                <td>{user.company.bs}</td>
                            </tr>
                        ))
                        }
                    </tbody>
                </table>
            </div>
        </div>
    );
};
export default Table;