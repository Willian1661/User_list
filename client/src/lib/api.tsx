import User from "../interfaces/user";

const getData = async () => {
  const response = await fetch("https://jsonplaceholder.typicode.com/users");

  if (!response.ok) {
    throw new Error("Couldn't fetch any data!");
  }

  const data: User[] = await response.json();

  return data;

};
export default getData;