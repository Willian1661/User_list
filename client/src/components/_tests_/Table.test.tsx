import React from "react";
import Table from "../Table";
import { getData } from "../../lib/api";
import { render, waitFor } from '@testing-library/react';

jest.mock('../../lib/api', () => ({
    getData: () => Promise.resolve([
        {
            id: 1,
            name: "Leanne Graham",
            username: "Bret",
            email: "Sincere@april.biz",
            address: {
                street: 'Kulas Light',
                suite: 'Apt. 556',
                city: 'Gwenborough',
                zipcode: '92998-3874',
                geo: {
                    lat: '-37.3159',
                    lng: '81.1496',
                }
            },
            phone: "1-770-736-8031 x56442",
            website: "hildegard.org",
            company: {
                name: 'Romaguera-Crona',
                catchPhrase: 'Multi-layered client-server neural-net',
                bs: 'harness real-time e-markets'
            }
        },
    ])
})
);

describe('Table component', () => {
    it('is rendenring Table component', async () => {

        const { getByText } = render(<Table data={getData()} title="foo" />)

        await waitFor(() => getByText("Leanne Graham"))

        expect(getByText('foo')).toBeTruthy();
        expect(getByText('Leanne Graham')).toBeTruthy();
    });
});