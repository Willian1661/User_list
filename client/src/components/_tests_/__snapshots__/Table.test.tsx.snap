// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component is rendenring Table component 1`] = `
<div
  className="flex items-center h-screen"
>
  <div
    className=" px-1 min-w-max"
  >
    <table
      className=" border-collapse border-2 border-solid"
      id="table"
    >
      <caption
        className=" text-2xl"
      >
        foo
      </caption>
      <thead>
        <tr>
          <th>
            ID
          </th>
          <th>
            Name
          </th>
          <th>
            UserName
          </th>
          <th>
            Email
          </th>
          <th>
            phone
          </th>
          <th>
            website
          </th>
          <th>
            address(city)
          </th>
          <th>
            address(zipcode)
          </th>
          <th>
            address(street)
          </th>
          <th>
            address(suite)
          </th>
          <th>
            address geo(lat)
          </th>
          <th>
            address geo(lng)
          </th>
          <th>
            company(name)
          </th>
          <th>
            company(catchPhrase)
          </th>
          <th>
            company(bs)
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
